commit a07961c98d2b6a4019b845b0d71c9455be05a5d0 (refs/changes/61/5061/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-11-14T13:44:57-08:00 (11 months ago)
    
    TRITON-960 CNAPI's ImageGet endpoint crashes CNAPI
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/endpoints/images.js b/lib/endpoints/images.js
index 537dcec..b0ecf19 100644
--- a/lib/endpoints/images.js
+++ b/lib/endpoints/images.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -66,7 +66,7 @@ Image.get = function handlerImageGet(req, res, next) {
     }, imageLoadTimeoutSeconds * 1000);
 
     req.stash.image.get(
-        { reqId: req.getId() },
+        { req_id: req.getId() },
         function (error, image) {
             clearTimeout(timeout);
 
diff --git a/lib/models/image.js b/lib/models/image.js
index 40fb1b7..c889fb8 100644
--- a/lib/models/image.js
+++ b/lib/models/image.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -13,6 +13,7 @@
  * with the intent of getting information about installed images.
  */
 
+var assert = require('assert-plus');
 var async = require('async');
 var sprintf = require('sprintf').sprintf;
 var util = require('util');
@@ -57,6 +58,9 @@ ModelImage.init = function (app) {
 ModelImage.prototype.get = function (opts, callback) {
     var self = this;
 
+    assert.object(opts, 'opts');
+    assert.uuid(opts.req_id, 'opts.req_id');
+
     ModelServer.get(self.serverUuid, function (err, servermodel, server) {
         var request = {
             task: 'image_get',
diff --git a/package.json b/package.json
index 01f2fab..6ed3965 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.13.1",
+  "version": "1.14.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
